hibernate-core/src/main/java/org/hibernate/dialect/FirebirdDialect.java

```diff
@@ -216,27 +215,16 @@ public SqlAstTranslatorFactory getSqlAstTranslatorFactory() {
 	/**
 	 * Firebird 2.5 doesn't have a real {@link java.sql.Types#BOOLEAN}
 	 * type, so...
+	 * @param from
+	 * @param to
 	 */
 	@Override
 	public String castPattern(CastType from, CastType to) {
-		if ( to==BOOLEAN
-				&& (from==LONG || from==INTEGER)) {
-			return "(0<>?1)";
-		}
-		if ( getVersion() < 300 ) {
-			if ( to==BOOLEAN && from==STRING ) {
-//				return "iif(lower(?1) similar to 't|f|true|false', lower(?1) like 't%', null)";
-				return "decode(lower(?1),'t',1,'f',0,'true',1,'false',0)";
-			}
-			if ( to==STRING && from==BOOLEAN ) {
-				return "trim(decode(?1,0,'false','true'))";
-			}
-		}
-		else {
-			if ( from==BOOLEAN
-					&& (to==LONG || to==INTEGER)) {
-				return "decode(?1,false,0,true,1)";
-			}
```

Comment:
It's at least a little bit sad that we're now using `case`/`when` for this on databases which have the much less verbose `decode()`.

Reply:
I'll readd that as good as I can